### PR TITLE
SNT-437 scenario form year range slider

### PR DESCRIPTION
**Pull request by Bewi**
## What problem is this PR solving?

Change year input when creating / editing a scenario to a slider

### Related JIRA tickets

SNT-437

## Changes

- Moved slider from impact to a component
- Use the slider in both impact and scenario form

## How to test

Go to SNT, create a scenario, play with the slider. 

Edit the scenario, play with the slider.

## Print screen / video

image

## Notes

N/A

## Doc

N/A

### CODE REVIEW

File: js/src/domains/scenarios/hooks/useScenarioFormState.tsx [hunk 1]
```diff
@@ -13,12 +13,17 @@ export type ScenarioFormValues = {
     end_year?: number;
 };
 
+export const SCENARIO_YEAR_RANGE = {
+    min: 2024,
+    max: 2035,
+} as const;
```
Comment (michiwend) [inline js/src/domains/scenarios/hooks/useScenarioFormState.tsx:19]: Two things:

We define this range now in the front-end and the back-end:
https://github.com/BLSQ/snt-malaria/blob/develop/api/scenarios/serializers.py#L41-L42

Maybe these could be exposed by the backend and then the second point is whether we want make them configurable or generate them based on current year?

File: js/src/domains/scenarios/hooks/useScenarioFormState.tsx
Comment (michiwend) [inline js/src/domains/scenarios/hooks/useScenarioFormState.tsx:98]: A bit picky and pre-existing but now with the min/max range of the slider this might become a problem when editing an old scenario that falls out of the current range. It would just pass the scenario's years but then the form would limit the slider to the current min/max range. So lets say we have an old scenario with 2020 but the new min is 2024. The form would show 2024 as min but the underlying value would be 2020 (see screenshot)

Image

File: js/src/domains/scenarios/hooks/useScenarioFormState.tsx [hunk 2]
```diff
@@ -13,12 +13,17 @@ export type ScenarioFormValues = {
     end_year?: number;
```
Comment (michiwend) [inline js/src/domains/scenarios/hooks/useScenarioFormState.tsx]: you can drop the ? now that we don't have the inputs that allowed blank and then remove null checks here https://github.com/BLSQ/snt-malaria/pull/271/changes#diff-ac48c305df74441b90f406dee5ce6ee83ea260dda6c681dc435c899526e4856cR45-R48

**Review by michiwend — CHANGES_REQUESTED** [review]
Tested locally, its so much easier to use than before. I found a few things worth checking:

These are not used anymore:
https://github.com/BLSQ/snt-malaria/blob/develop/js/src/domains/messages.ts#L788-L791
https://github.com/BLSQ/snt-malaria/blob/develop/js/src/domains/messages.ts#L364-L372

File: js/src/domains/scenarios/hooks/useScenarioFormState.tsx
Comment (Bewi) [reply js/src/domains/scenarios/hooks/useScenarioFormState.tsx:98]: Range was always this, not sure how you got a scenario with a start date lower than 2024

File: js/src/domains/scenarios/hooks/useScenarioFormState.tsx (hunk 1 above)
Comment (Bewi) [reply js/src/domains/scenarios/hooks/useScenarioFormState.tsx:19]: We can discuss about it but I won't fix it now, this was just to change the input.
To be honnest, I don't like the min max, but it requires some thinking and validation from Fred

File: js/src/domains/scenarios/hooks/useScenarioFormState.tsx (hunk 2 above)
Comment (Bewi) [reply js/src/domains/scenarios/hooks/useScenarioFormState.tsx]: Cleaned it up

File: js/src/domains/scenarios/hooks/useScenarioFormState.tsx
Comment (michiwend) [reply js/src/domains/scenarios/hooks/useScenarioFormState.tsx:98]: I did that in django admin just to test this edge case. I don't think it is very critical but it might fall on our feat later if we allow the min/max range to change

**Review by michiwend — APPROVED** [review]
